services/QuillLMS/app/controllers/teachers/classroom_manager_controller.rb

```diff
@@ -135,14 +135,27 @@ def getting_started
   end
 
   def scores
-    scores = Scorebook::Query.run(params[:classroom_id], params[:current_page], params[:unit_id], params[:begin_date], params[:end_date], current_user.utc_offset)
+    classroom = Classroom.find(params[:classroom_id])
+    cache_groups = {
+      unit: params[:unit_id],
+      page: params[:current_page],
+      begin: params[:begin_date],
+      end: params[:end_date],
+      offset: current_user.utc_offset
+    }
 
-    last_page = scores.length < 200
+    json = current_user.classroom_cache(classroom, key: 'classroom_manager.teacher_dashboard_metrics', groups: cache_groups) do
```

Comment:
I don't think this key can be right - shouldn't that be for the metrics that the teacher sees on their dashboard?